tool/tsh/kube.go

```diff
@@ -312,11 +314,27 @@ func fetchKubeStatus(ctx context.Context, tc *client.TeleportClient) (*kubernete
 		return nil, trace.Wrap(err)
 	}
 
-	k8host, _ := tc.KubeProxyHostPort()
-	kubeStatus.tlsServerName = addSubdomainPrefix(k8host, alpnproxy.KubeSNIPrefix)
+	if tc.TLSRoutingEnabled {
+		kubeStatus.tlsServerName = getKubeTLSServerName(tc)
+	}
+
 	return kubeStatus, nil
 }
 
+// getKubeTLSServerName returns k8 server name used in KUBECONFIG to leverage TLS Routing.
+func getKubeTLSServerName(tc *client.TeleportClient) string {
+	webHost := tc.WebProxyHost()
```

Comment:
Should we use `KubeProxyHost()` here instead? In case a different Kubernetes public_addr is specified in the config. It will fallback to `WebProxyHost()` anyway if it's missing, right?

Reply:
yes, after aligning the `TSH client config`  https://github.com/gravitational/teleport/pull/8940/files#diff-82cf173562bffe0bb2951ce3cb64f9af18c8aedd820b62026f11be2b17690429R2666 the `KubeProxyHost` should be used here.